### PR TITLE
ENH: Vizualisation of decoding results

### PR DESCRIPTION
**Pull request by cmoutard**
With matplotlib it is sometimes painful to modify an already-made object, and here I think the rule 'prevention is better than cure' may apply.
In order to have flexibility in the design of the figures associated with decoding results, this PR proposes to have some optional arguments in plot_gat_matrix() and plot_gat_diagonal() to adapt their format as needed.
Here I itemize the propositions for each function :

## plot_gat_matrix()

* The colorbar becomes optional
* xlabel/ylabel become optional
* Return also the variable 'im' as it used to be in a former version of the code

## plot_gat_diagonal()

* Legend becomes optional, and the argument -if not None- indicates the location of the box
* xlim (xmin and xmax) can be specified
* xlabel/ylabel become optional
* [FIX] The argument 'title', proposed but not implemented, is made operational

## Example

These enhancements come from some personal needs which some users may encounter. For this reason, the options proposed are not exhaustive and biased by my personnal experience. It may be improved in the future.
To illustrate how practical this is, here is some fresh results I got from the recently-merged-good-old PR/1629. I take the opportunity to congrat the decoding team, great work !! Following, what one can do many thanks to you :

decoding_example

### CODE REVIEW

File: mne/viz/decoding.py [hunk 1]
```diff
@@ -115,12 +139,19 @@ def plot_gat_diagonal(gat, title=None, ymin=0., ymax=1., ax=None, show=True,
...
-    ax.set_xlabel('Time (s)')
-    ax.set_ylabel('Classif. score ({0})'.format(
-        'AUC' if 'roc' in repr(gat.scorer_) else r'%'
-    ))
-    ax.legend(loc='best')
+    if xmin is not None and xmax is not None:
+        ax.set_xlim(xmin, xmax)
+    if xlabel:
+        ax.set_xlabel('Time (s)')
+    if ylabel:
```
Comment (dengemann) [inline mne/viz/decoding.py]: we should use explicit testing here. Like is not None

Comment (dengemann) [inline mne/viz/decoding.py]: ... or is not False

File: mne/viz/decoding.py [hunk 2]
```diff
@@ -37,11 +38,19 @@ def plot_gat_matrix(gat, title=None, vmin=0., vmax=1., tlim=None,
...
+        If True, the xlabel is displayed. Defaults to True.
+    ylabel : bool
+        If True, the ylabel is displayed. Defaults to True.
 
     Returns
     -------
     fig : instance of matplotlib.figure.Figure
         The figure.
+    im : instance of matplotlib.image.AxesImage
+        The image to be displayed.
```
Comment (agramfort) [inline mne/viz/decoding.py]: wrong returned parameters order. You seem to return im and then fig

**Comment by agramfort** [comment]
besides LGTM

**Comment by agramfort** [comment]
LGTM

@dengemann @kingjr are you happy too?

**Comment by kingjr** [comment]
Thanks Clément! Sorry for the delay, I had missed this issue.

In the same line of thoughts, it would be good to make the  default plot_gat_matrix color limits defined according to min and max of the plotted scores (by default it's 0 and 1, but this is only valid for scorer that are bounded between these two values).

Regarding the fig, axis, colorbar, legend and img specifications, I think we should maximize the possibility to pass all possible options to the matplotlib functions using **kwargs. See the comments in the commits.

Thanks again.

**Comment by agramfort** [comment]
-1 on kwargs. It makes options impossible to find interactively by users.

please please address the use case of the 90% of users and don't think to
allow every possible option to be a generic/powerful/blah software !!!

**Comment by kingjr** [comment]
My point is that we should have dictionaries for ```colorbar``` and ```legend```, as it is more explicit and more convenient for future enhancements. Typically, I can already see that legend = int to define the legend location will be annoying at some point, because it's likely that we'll soon want to specify some other parameters of the legend. 

We can still explicit the dicts keys that we think are relevant; the doc would be similar to what we're doing now anyway. That would address the 90% use case, and allows advanced use cases for free.

**Comment by agramfort** [comment]
you'll "soon want/need it".

Please read:

http://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it

let us add features when actually deemed needed.

**Comment by kingjr** [comment]
Ok, your call, although I still find it weird to have legend is int for a legend location, a True | False would address the 90% cases.

**Comment by agramfort** [comment]
>
>  a True | False would address the 90% cases.
>
good call. Let's have it to true or false then :)

**Comment by cmoutard** [comment]
I completely agree on the strangeness of legend=int (what I really tried to avoid). But then how to adjust the location of the legend ? Because the default loc="best" is performing poorly, the legend overlapping the curves while having a lot of free space in some other parts of the figure.
The only manner I have found to change the location retrospectively (maybe there are others) is there and it's not really nice since it implies to draw the figure :
http://stackoverflow.com/questions/23238041/move-and-resize-legends-box-in-matplotlib
Any solutions are welcome. ThX !

**Comment by dengemann** [comment]
voting for the most simple API. Everythinh is an object in matplotlib. You can update any sub-object of a figure post-hoc. Cf. mpl.findobj

**Comment by cmoutard** [comment]
ok, thX

**Comment by cmoutard** [comment]
@dengemann @kingjr 
So I have made legend a boolean argument and have deleted the unnecessary "return im".
Is it all fine now ?

**Comment by dengemann** [comment]
looks good to me! Travis issues seem unrealted. Please add your name to the source files you've edited.

**Comment by kingjr** [comment]
Yep, thanks Clément!

**Comment by agramfort** [comment]
thanks @cmoutard !

**Pull request merged**